dynamiclinks/app/src/main/java/com/google/firebase/quickstart/deeplinks/java/MainActivity.java

```diff
@@ -165,4 +181,39 @@ private void validateAppCode() {
                     .create().show();
         }
     }
+
+    @VisibleForTesting
+    public void buildShortLinkFromParams(@NonNull Uri deepLink, int minVersion) {
+      String uriPrefix = getString(R.string.dynamic_links_uri_prefix);
+
+      // Set dynamic link parameters:
+      //  * URI prefix (required)
+      //  * Android Parameters (required)
+      //  * Deep link
+      FirebaseDynamicLinks.getInstance()
+          .createDynamicLink()
+          .setDomainUriPrefix(uriPrefix)
+          .setAndroidParameters(
+             new AndroidParameters.Builder()
+                 .setMinimumVersion(minVersion)
+                 .build()
+          )
+          .setLink(deepLink)
+          .buildShortDynamicLink()
+          .addOnCompleteListener(this, new OnCompleteListener<ShortDynamicLink>() {
+            @Override
+            public void onComplete(@NonNull Task<ShortDynamicLink> task) {
+              if (task.isSuccessful()) {
+                // Short link created
+                Uri shortDynamicLink = task.getResult().getShortLink();
+                TextView shortLinkTextView = findViewById(R.id.shortLinkViewSend);
+                shortLinkTextView.setText(shortDynamicLink.toString());
+              } else {
+                // Error
+                Log.e(TAG, task.getException().getMessage());
+                throw new Error(task.getException().getMessage());
```

Comment:
Simply logging the error should be fine, there's no need to re-throw it.
That's the approach we use across all quickstarts.

```suggestion
                Log.e(TAG, task.getException().getMessage());
```

Reply:
Thanks for the comment. I removed the throw statements